diff --git a/node_modules/escodegen/escodegen.js b/node_modules/escodegen/escodegen.js
index 847fc37..dc9d780 100644
--- a/node_modules/escodegen/escodegen.js
+++ b/node_modules/escodegen/escodegen.js
@@ -2636,7 +2636,7 @@
 
     FORMAT_DEFAULTS = getDefaultOptions().format;
 
-    exports.version = require('./package.json').version;
+    // exports.version = require('./package.json').version;
     exports.generate = generate;
     exports.attachComments = estraverse.attachComments;
     exports.Precedence = updateDeeply({}, Precedence);
