diff --git a/node_modules/gallinago/src/lib/runner.js b/node_modules/gallinago/src/lib/runner.js
index 161e626..0cada58 100644
--- a/node_modules/gallinago/src/lib/runner.js
+++ b/node_modules/gallinago/src/lib/runner.js
@@ -1,5 +1,5 @@
 import fs from 'fs';
-import os from 'os';
+import os, { type } from 'os';
 import path from 'path';
 import { spawn, spawnSync } from 'child_process';
 
@@ -17,9 +17,9 @@ class Runner {
     return new Promise((resolve, reject) => {
       if (path.isAbsolute(rootDir)) {
 
-        if (!fs.existsSync(rootDir)) {
-          fs.mkdirSync(rootDir);
-        }
+        // if (!fs.existsSync(rootDir)) {
+        //   fs.mkdirSync(rootDir);
+        // }
 
         this.rootDir = rootDir;
 
@@ -37,7 +37,7 @@ class Runner {
     });
   }
 
-  runCommand(binPath, args = null, options = {}) {
+  runCommand(binPath, args, options = {}) {
     return new Promise((resolve, reject) => {
       const executable = 'node';
       const isWindows = os.platform() === 'win32';
@@ -47,10 +47,14 @@ class Runner {
       const spawnAction = options.async ? spawn : spawnSync;
       let err = '';
 
-      if (args) {
+      if (Array.isArray(args)) {
+        finalArgs.push(...args);
+      } else if(typeof args === 'string') {
         finalArgs.push(args);
       }
 
+      // console.log({ args, forwardArgs, finalArgs });
+
       if (!fs.existsSync(binPath)) {
         reject(`Error: Cannot find path ${binPath}`);
       }
