diff --git a/node_modules/gallinago/src/lib/runner.js b/node_modules/gallinago/src/lib/runner.js
index b57fb8b..1944dd3 100644
--- a/node_modules/gallinago/src/lib/runner.js
+++ b/node_modules/gallinago/src/lib/runner.js
@@ -87,8 +87,11 @@ class Runner {
   stopCommand() {
     return new Promise((resolve) => {
       if (this.childProcess) {
+        this.childProcess.on('exit', () => {
+          resolve();
+        })
+
         this.childProcess.kill();
-        resolve();
       } else {
         resolve()
       }
